### PR TITLE
feat(deploy/ha): Allow disabling of clouddriver-ro-deck.

### PR DESCRIPTION
**Pull request by joonlim**

### CODE REVIEW

**Comment by joonlim** [comment]
@jtk54

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java [hunk 1]
```diff
@@ -30,15 +30,24 @@
...
-    ).filter(e -> !StringUtils.isBlank(e))
-        .collect(Collectors.toList());
+    boolean redisMasterEndpointIsBlank = StringUtils.isBlank(clouddriverHaService.getRedisMasterEndpoint());
+    boolean redisSlaveEndpointIsBlank = StringUtils.isBlank(clouddriverHaService.getRedisSlaveEndpoint());
+    boolean redisSlaveDeckEndpointIsBlank = StringUtils.isBlank(clouddriverHaService.getRedisSlaveDeckEndpoint());
 
-    if (redisEndpoints.size() != 3 && redisEndpoints.size() != 0) {
-      p.addProblem(Problem.Severity.ERROR, "Please provide values for Clouddriver Redis master endpoint, Redis slave endpoint, and Redis slave-deck endpoint or leave them all blank.");
+    if (clouddriverHaService.isDisableClouddriverRoDeck()) {
+      if (redisMasterEndpointIsBlank && redisSlaveEndpointIsBlank) {
```
Comment (jtk54) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: What's the rationale behind continuing if there are no redii configured, but failing if just one isn't specified?

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java [hunk 2]
```diff
@@ -30,15 +30,24 @@
...
+        return;
+      }
+      if (redisMasterEndpointIsBlank || redisSlaveEndpointIsBlank) {
+        p.addProblem(Problem.Severity.ERROR, "Please provide values for Clouddriver Redis master endpoint and Redis slave endpoint, or leave them both blank.");
+      }
+    } else {
+      if (redisMasterEndpointIsBlank && redisSlaveEndpointIsBlank && redisSlaveDeckEndpointIsBlank) {
```
Comment (jtk54) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: Same question here.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java (hunk 1 above)
Comment (joonlim) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: Any services without a redis endpoint configured will use the redis deployed by halyard, which is acceptable. However, spinnaker will be broken if there is at least one clouddriver pointing to an external redis and at least one clouddriver still pointing to the redis that is deployed by halyard.

Comment (joonlim) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: i.e.,

clouddriver-rw and clouddriver-caching points to external redis master
clouddriver-ro points to halyard-deployed redis.
^ fails bc clouddriver-ro can't read clouddriver data from redis master

clouddriver-rw and clouddriver-caching points to halyard-deployed redis
clouddriver-ro points to external redis.
^ fails bc clouddriver-ro again can't read clouddriver data from halyard-deployed redis

Comment (jtk54) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: SG, just wanted to make sure I wasn't missing something.

Comment (joonlim) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/ha/ClouddriverHaServiceValidator.java]: Thanks, could you approve this?

**Pull request merged**